src/core/FollowSearches.js

```
@@ -14,110 +19,116 @@ import { getToken } from "./token.js";
 class FollowSearches {
   constructor() {
     this.token = getToken();
+    this.baseUrl = "https://stage.followsearches.dandigbib.org";
   }
   /**
    * Get all the users searches.
+   * /list/{listName}
    *
-   * @param {string} listName
+   * @param {object} options
+   * @param {string} options.listName
+   * @param {number} options.page min: 1 - Has no effect if not set.
+   * @param {number} options.size min: 1 - If not set returns the full list.
    * @returns {Promise<Search[]>} a list of searches.
    * @memberof FollowSearches
    */
-  getSearches(listName) {
-    return new Promise((resolve, reject) => {
-      console.info(`Getting searches: ${listName}`);
-      try {
-        setTimeout(() => {
-          console.info("The search data is returned.");
-          resolve([
-            {
-              searchTitle: "Harry Potter stuff",
-              searchQuery: "harry potter"
-            },
-            {
-              searchTitle: "Star wars stuff",
-              searchQuery: "star wars"
-            },
-            {
-              searchTitle: "Marvel stuff",
-              searchQuery: "avengers"
-            }
-          ]);
-        }, 500);
-      } catch (err) {
-        reject("Unspecified error.");
+  async getSearches({ listName = "default", page, size } = {}) {
+    if (!listName) {
+      throw Error("listName must be provided.");
+    }
+
+    const query = [page && `page=${page}`, size && `size=${size}`]
+      .filter(parameter => parameter)
+      .join("&");
+
+    const raw = await fetch(
+      `${this.baseUrl}/list/${listName}${query ? "?" + query : ""}`,
+      {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          Authorization: `Bearer ${this.token}`
+        }
       }
-    });
+    );
+
+    return await raw.json();
```

Have you considered error handling as a part of these methods or is the Follow Searches error reporting so generic that we cannot do much better than let the promise fail?

---

I would rather have the consumer handle the errors as it is right now.

As far as I could see FollowSearches fails with status codes mainly. Which might as well be handled by the consumer as in the implementation. Handling in the implementation right now would generate double the code because the consumer need to handle and surface their errors as well.